SIPS/sip-20.md

```diff
@@ -1,189 +1,237 @@
 
 ```typescript
-/**
- * A WebSocket connection as specified in the Snap manifest.
- *
- * @property type - The type of connection. Currently, only WebSockets are
- * supported.
- * @property url - The URL to connect to. This must be a valid WebSocket URL,
- * starting with `wss://`. URLs starting with `ws://` are not allowed.
- * @property dataType - The type of data expected from the WebSocket. If this
- * field is omitted, text is assumed.
- */
-type WebSocketConnection = {
-  type: 'websocket';
+
+type OpenWebSocketParams = {
   url: string;
-  dataType?: 'text' | 'binary';
 };
+```
+The RPC method takes one parameter:
+
+- `url` - The URL of the WebSocket service to connect to. 
+  - The URL MUST be a valid WebSocket URL, starting with `wss://`. URLs starting with `ws://` are not allowed.
+  - Only one WebSocket connection can be opened per URL at a time. If a Snap tries to open a new connection to the same URL while an existing connection is still open, the client MUST throw an error.
+
+An example of usage is given below.
+
+```typescript
+snap.request({
+  method: "snap_openWebSocket",
+  params: {
+    url: "wss://example.com/websocket",
+  },
+});
 
-/**
- * An external data connection as specified in the Snap manifest.
- *
- * Currently, only {@link WebSocketConnection} is supported.
- */
-type ExternalDataConnection = WebSocketConnection;
-
-/**
- * External data connections as specified in the Snap manifest.
- *
- * This is the value of the `endowment:external-data` field in the Snap
- * manifest.
- *
- * @property maxRequestTime - The maximum request time for the `onExternalData`
- * entry point, as described in SIP-21.
- * @property connections - The external data connections.
- */
-type ExternalData = {
-    maxRequestTime?: number;
-    connections: ExternalDataConnection[]
+```
+
+#### `snap_closeWebSocket`
+This method allows a Snap to close an existing WebSocket connection. The method takes the unique identifier of the connection as a parameter.
+
+```typescript
+type CloseWebSocketParams = {
+  id: string;
 };
 ```
+The RPC method takes one parameter:
+- `id` - The unique identifier of the WebSocket connection to close. This identifier is returned by the `snap_openWebSocket` method.
 
-#### Example
-
-The new field can be specified as follows in a `snap.manifest.json` file:
-
-```json
-{
-  "initialPermissions": {
-    "endowment:external-data": {
-      "maxRequestTime": 50000,
-      "connections": [
-        {
-          "type": "websocket",
-          "url": "wss://example.com/binary/endpoint",
-          "dataType": "binary"
-        },
-        {
-          "type": "websocket",
-          "url": "wss://example.com/text/endpoint",
-          "dataType": "text"
-        }
-      ]
-    }
-  }
-}
+An example of usage is given below.
+
+```typescript
+snap.request({
+  method: "snap_closeWebSocket",
+  params: {
+    id: "unique-connection-id",
+  },
+});
 ```
+#### `snap_sendMessage`
```

Comment:
I think this name is too generic, especially given that everything else is specific to WebSockets.